### DISCUSSION TITLE
Is this an anti-pattern?

### DISCUSSION THREAD

**Discussion by fazulk** [vuejs/pinia#1763] [Help and Questions] [answered]
Not too versed in how pinia is working under the hood. But could this be considered anti pattern or slow down peformance in any way? 

```
computed: {
    availableStates() {
      return useGlobalStore().appConfig?.states
    },
 },
 ```

**Comment by posva** [comment, marked as answer]
yes, you could say it's an anti pattern for composables in general: they are meant to be used in setup and script setup only. If you want to use the option api with pinia, use the map helpers
